jmclient/jmclient/configure.py

```diff
@@ -748,6 +806,11 @@ def load_program_config(config_path="", bs=None, plugin_services=[]):
             if not os.path.exists(plogsdir):
                 os.makedirs(plogsdir)
             p.set_log_dir(plogsdir)
+    # Check if a onion message channel was configured, and if so,
+    # check there is only 1; multiple directory nodes will be inside the config.
+    chans = get_mchannels()
+    onion_chans = [x for x in chans if "type" in x and x["type"] == "onion"]
+    assert len(onion_chans) < 2
```

Comment:
Can we do this check  inside `get_mchannels()`? Here we have to loop again through this list, but we could've asserted there is only one onion channel already.

Reply:
Done in https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/1182/commits/09c510742dbf4ffd23502289bfba7f096bd5bd43